### PR TITLE
Clarify steps needed for Cloud SQL

### PR DESCRIPTION
**Pull request by daxmc99**
This adds additional clarification for how to access our production databases

Followup to https://sourcegraph.slack.com/archives/D012ZE3Q5J7/p1604332857000100

### CODE REVIEW

File: handbook/engineering/deployments/postgresql.md [hunk 1]
```diff
@@ -1,25 +1,70 @@
...
+
+There are two ways of connecting: either using the `gcloud beta sql connect` command or running the proxy on a port locally.
+
+### Command line only use (pgsql)
+
+You may use these gcloud commands to connect directly to the databases:
+
+- Default db {[Password](https://my.1password.com/vaults/dnrhbauihkhjs5ag6vszsme45a/allitems/svfiw4vcbxhhbobpl442olyebu/)}
+  ```
+    gcloud beta sql connect --project sourcegraph-dev sg-cloud-732a936743 --user=sg -d=sg
```
Comment (nicksnyder) [inline handbook/engineering/deployments/postgresql.md:25]: Doesn't work for me

```
% gcloud beta sql connect --project sourcegraph-dev sg-cloud-732a936743 --user=sg -d=sg
ERROR: (gcloud.beta.sql.connect) Cloud SQL Proxy could not be found in PATH. See https://cloud.google.com/sql/docs/mysql/sql-proxy#install for information on installing.
```

File: handbook/engineering/deployments/postgresql.md [hunk 2]
```diff
@@ -1,25 +1,70 @@
...
+https://console.cloud.google.com/bigquery?sq=527047051561:67f2616f4acb4b7cb3639e4a97e2f4aa
+) and their `EXTERNAL_QUERY` syntax.
+
+There are two ways of connecting: either using the `gcloud beta sql connect` command or running the proxy on a port locally.
+
+### Command line only use (pgsql)
+
+You may use these gcloud commands to connect directly to the databases:
+
+- Default db {[Password](https://my.1password.com/vaults/dnrhbauihkhjs5ag6vszsme45a/allitems/svfiw4vcbxhhbobpl442olyebu/)}
```
Comment (nicksnyder) [inline handbook/engineering/deployments/postgresql.md:23]: how/where is the password provided?

File: handbook/engineering/deployments/postgresql.md [hunk 3]
```diff
@@ -1,25 +1,70 @@
...
 You can also directly view database in [GCP](https://console.cloud.google.com/sql/instances?project=sourcegraph-dev)
 
-We currently run two seperate databases. The `sg-cloud` database is the primary database and the `sg-cloud-code-intel` is used by the code-intel team.
+We utilize the GCP utility of the [Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy) to connect to our production databases. By default, our Cloud SQL databases are not accessible without this proxy.
+
+For read-only access there is also an option of using [BigQuery](
+https://console.cloud.google.com/bigquery?sq=527047051561:67f2616f4acb4b7cb3639e4a97e2f4aa
+) and their `EXTERNAL_QUERY` syntax.
+
+There are two ways of connecting: either using the `gcloud beta sql connect` command or running the proxy on a port locally.
```
Comment (nicksnyder) [inline handbook/engineering/deployments/postgresql.md:17]: As a developer, how do I know which way to choose?

File: handbook/engineering/deployments/postgresql.md [hunk 4]
```diff
@@ -1,25 +1,70 @@
...
 ```
 
+Now, in a new terminal run the command below, the database will be running on `localhost:5555`
+
+  ```
+    pgsql -h localhost -p 5555 -d sg -U sg
+  ```
+  _Note:_ You may need to set or unset your PGPASSWORD environment variable to the value of the password of the user you are using to connect (`-U sg`).
+
+If you are frequently using this to access the database for read-only operations, please let the #dev-ops channel know so we can add a read-only replica.
```
Comment (nicksnyder) [inline handbook/engineering/deployments/postgresql.md:48]: At the very least, it seems like we should create a readonly user.

File: handbook/engineering/deployments/postgresql.md (hunk 1 above)
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:25]: Ahh, I thought it prompted to install the beta utils if you did not have them 
```
You do not currently have this command group installed.  Using it
requires the installation of components: [beta]


Your current Cloud SDK version is: 310.0.0
Installing components from version: 310.0.0

┌─────────────────────────────────────────────┐
│     These components will be installed.     │
├──────────────────────┬────────────┬─────────┤
│         Name         │  Version   │   Size  │
├──────────────────────┼────────────┼─────────┤
│ gcloud Beta Commands │ 2019.05.17 │ < 1 MiB │
└──────────────────────┴────────────┴─────────┘

For the latest full release notes, please visit:
  https://cloud.google.com/sdk/release_notes

Do you want to continue (Y/n)?
```

Comment (nicksnyder) [reply handbook/engineering/deployments/postgresql.md:25]: What makes you think I don't have beta tools installed?

File: handbook/engineering/deployments/postgresql.md (hunk 3 above)
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:17]: gcloud is mainly for quick `pgsql` commands, the proxy is used when you want to connect an external database viewer or GUI program.

Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:17]: ^ I will update it with this info

Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:17]:
```suggestion
There are two ways of connecting: either using the `gcloud beta sql connect` command which will use the `pgsql` client or running the proxy on a port locally to utilize whatever tools you typically use.

```

File: handbook/engineering/deployments/postgresql.md (hunk 1 above)
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:25]: It seems like the cloud-sql-proxy is not installed as part of the beta tools. I thought it was.

File: handbook/engineering/deployments/postgresql.md (hunk 2 above)
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:29]: The login should prompt for it

File: handbook/engineering/deployments/postgresql.md (hunk 4 above)
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:54]: Added here https://github.com/sourcegraph/infrastructure/pull/2234

File: handbook/engineering/deployments/postgresql.md [hunk 5]
```diff
@@ -1,25 +1,72 @@
...
-You can also directly view database in [GCP](https://console.cloud.google.com/sql/instances?project=sourcegraph-dev)
+You can also directly view the database in [GCP](https://console.cloud.google.com/sql/instances?project=sourcegraph-dev)
 
-We currently run two seperate databases. The `sg-cloud` database is the primary database and the `sg-cloud-code-intel` is used by the code-intel team.
+We utilize the GCP utility of the [Cloud SQL Proxy](https://cloud.google.com/sql/docs/postgres/sql-proxy) to connect to our production databases. By default, our Cloud SQL databases are not accessible without this proxy.
 
-Use the cloud sql proxy to connect to the database
+For read-only access, there is also an option of using [BigQuery](
+https://console.cloud.google.com/bigquery?sq=527047051561:67f2616f4acb4b7cb3639e4a97e2f4aa
+) and their `EXTERNAL_QUERY` syntax.
```
Comment (bobheadxi) [inline handbook/engineering/deployments/postgresql.md:15]: I think using BigQuery might deserve a section of its own, with the full `EXTERNAL_QUERY` here (did not realize that it was just a stand-in for a table), e.g. seeing:

```sql
SELECT * FROM EXTERNAL_QUERY("...", "SELECT name::text,created_at::text FROM repo LIMIT 5;");
```

would be a lot more self-explanatory. Seems a lot less setup than the alternative, so for UI-and-mouse noobs like me I would probably go for this more often than not 😛

File: handbook/engineering/deployments/postgresql.md [hunk 6]
```diff
@@ -1,25 +1,72 @@
...
+
+  ```
+    export PGPASSWORD='<$PASSWORD>'
+    pgsql -h localhost -p 5555 -d sg -U 'dev-readonly'
+  ```
+
+## Example queries
+🔥 You are directly interfacing with the production database. If you are unsure of any commands, please reach out in #dev-chat or #dev-ops
+    Please prefer using the readonly user 'frontend-dev'
+- See all fields on a table (ie the `repo` table)
```
Comment (bobheadxi) [inline handbook/engineering/deployments/postgresql.md:56]: formatting nit:

```suggestion
## Example queries

> 🔥 You are directly interfacing with the production database. If you are unsure of any commands, please reach out in #dev-chat or #dev-ops.
> Please prefer using the readonly user `frontend-dev`

- See all fields on a table (ie the `repo` table)
```

**Comment by nicksnyder** [comment]
don't wait on my review, ultimately this documentation should make it clear to a new teammate how to connect to the database. The best review here is to have a new teammate walk through these steps and give you feedback on whether they were successful or not (feel free to merge first)

File: handbook/engineering/deployments/postgresql.md [hunk 7]
```diff
@@ -1,25 +1,74 @@
...
+
+There are two ways of connecting: either using the `gcloud beta sql connect` command, which will use the `pgsql` client, or running the proxy on a port locally to utilize your preferred tools.
+
+### Command line only use (pgsql)
+
+You may use these gcloud commands to connect directly to the databases:
+
+- Default db {[Password](https://start.1password.com/open/i?a=HEDEDSLHPBFGRBTKAKJWE23XX4&v=dnrhbauihkhjs5ag6vszsme45a&i=pjxf64qxwsin4d56xij6vm3gva&h=my.1password.com)}
+  ```
+    gcloud beta sql connect --project sourcegraph-dev sg-cloud-732a936743 --user=dev-readonly -d=sg
```
[the remark this answers is not in the document]
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:30]: Do you find this command useful or do you just use the proxy by itself?

File: handbook/engineering/deployments/postgresql.md [hunk 8]
```diff
@@ -1,25 +1,74 @@
...
+[Download](https://cloud.google.com/sql/docs/postgres/quickstart-proxy-test#install_and_authenticate_the_gcloud_command-line_tool) and setup the cloud sql proxy
 ```
   cloud_sql_proxy -instances=sourcegraph-dev:us-central1:sg-cloud-732a936743=tcp:5555
 ```
 
+Now, in a new terminal, run the command below. The database will be running on `localhost:5555`
+
+  ```
+    export PGPASSWORD='<$PASSWORD>'
+    pgsql -h localhost -p 5555 -d sg -U 'dev-readonly'
```
[the remark this answers is not in the document]
Comment (daxmc99) [reply handbook/engineering/deployments/postgresql.md:50]:
```suggestion
    psql -h localhost -p 5555 -d sg -U 'dev-readonly'
```

**Pull request merged**